In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import datetime as datetime

# Introduction to Digital Forensics - Assignment 3

Contributors (Project group 2):
- Carlos de Bourbon de Parme
- Vera Schockemöhle
- Floor Schutrups
- Jelle Hulter
- Nora Yigit
- Arda Öz

# 1. Introduction/Motivation
For this assignment, we have first read the two articles which were given to us from Brian Krebs. After that, we inspected the Excel file which was posted on Canvas. In this file, the `attacks` table from the database of TWBooter could be found. During this exercise, we would like to know how many attacks were performed and how many unique customers used this booter. Also, we would like to perform further analysis of this dataset.

# 2. Methodology

To start, we have converted the Excel file to a CSV. Then, we fixed the "victim" column by replacing all commas with a dot, because many records were formatted wrongly.

In [2]:
df = pd.read_csv("booter.csv", delimiter=';')
df['victim'] = df['victim'].str.replace(',', '.')
df

id               processid                  victim  \
0          1                    5672            91.214.20.52   
1          2                   29065            91.214.20.52   
2          3                   23005             78.31.8.124   
3          4                   29411          199.204.44.188   
4          5                   19298  http://chrome-mods.com   
...      ...                     ...                     ...   
48768  48840  15874|1992|28878|32327         109.234.152.119   
48769  48841       15022|19087|28887          94.242.205.227   
48770  48842       15026|19091|28896   http://94.242.205.227   
48771  48843   15898|3409|32693|2010   http://94.242.205.227   
48772  48844  15907|3418|32702|20944          94.242.205.227   

       port_service`_type    type  duration        time  stopped  \
0                      80   ESSYN        33  1358989954        1   
1                      80     UDP      1783  1358990329        1   
2                      80   ESSYN       393  1358991360        0   
3                      80   ESSYN        60  1358995416        0   
4                      80    RUDY     50000  1358995471        1   
...                   ...     ...       ...         ...      ...   
48768                1973   ESSYN       300  1363351101        0   
48769                  80  SOURCE        65  1363351473        1   
48770                  80    ARME        65  1363351493        1   
48771                  80   ESSYN        65  1363351507        1   
48772                  80   ESSYN        65  1363351516        0   

       client_attacker                           server_used_in_th_attack  
0                    1                                       94.102.49.76  
1                    1                                       93.174.93.30  
2                    1                                      89.248.172.97  
3                    4                                       93.174.93.30  
4                    3                                        72.9.144.80  
...                ...                                                ...  
48768              258  94.102.49.79|94.102.51.252|89.248.160.146|89.2...  
48769              261          72.9.144.80|67.222.156.241|89.248.160.146  
48770              261          72.9.144.80|67.222.156.241|89.248.160.146  
48771              261  94.102.49.79|89.248.172.96|89.248.162.242|94.1...  
48772              261  94.102.49.79|89.248.172.96|89.248.162.242|89.2...  

[48773 rows x 10 columns]

## 1. The total number of attack records?


In [3]:
len(df)

48773

## 2. Number of unique users?

In [4]:
len(df['client_attacker'].unique())

277

## 3. Top 10 victim IP addresses ?

In [5]:
is_ip = df['victim'].str.match("^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$")
ips = df[is_ip]
ips['victim'].value_counts().head(3)

176.31.90.154    280
72.9.144.6       267
198.1.122.66     198
Name: victim, dtype: int64

## 4. Top 10 victim domain names?

In [6]:
domains = df[np.invert(is_ip)]
domains['victim'].value_counts().head(3)

http://87.106.136.156     1106
http://212.7.214.197       798
http://www.relog.co.il     635
Name: victim, dtype: int64

## 5. The longest attack duration?

In [7]:
max(df['duration'])

500000

## 6. Top 3 attack type?

In [8]:
df['type'].value_counts().head(3)

ESSYN    19618
UDP      10505
RUDY      5808
Name: type, dtype: int64

## 7. Which owner id performed most of the attacks?

In [9]:
df['client_attacker'].value_counts().head(1)

272    2307
Name: client_attacker, dtype: int64

## 8. How many attacks 'krebsonsecurity.com' suffered?

In [10]:
len(df[df['victim'].str.contains('krebsonsecurity.com')])

15

## 9. Add two statistics that I didn't list above.

Average amount of servers used per attack:

In [11]:
df['server_used_in_th_attack'].str.split('|').apply(lambda x: len(x) if isinstance(x, list) else 0).mean()

2.725155311340291

Top 3 most attacked ports:

In [12]:
df['port_service`_type'].value_counts().head(3)

80      37571
6005     2058
6000      630
Name: port_service`_type, dtype: int64

# 3. Conclusion
We have learnt what the database of a booter looks like and we have gotten a better idea about how these Denial of Service attacks are performed in practice. At the start, we mentioned that our goal was to perform further analysis on this dataset with two characteristics in particular.

## How many attacks have been performed?

This was answered in section 2.1: 48773

## How many unique customers had this booter?
This was answered in section 2.2: 277